lpcompiler.pas

```diff
@@ -2697,10 +2697,7 @@ function TLapeCompiler.ParseExpression(ReturnOn: EParserTokenSet = []; FirstNext
                 _LastNode := _Var
               else
               begin
-                if (Method is TLapeTree_InternalMethod) and
-                   TLapeTree_InternalMethod(Method).ForceParam and
-                   (not (Tokenizer.Tok in ReturnOn))
-                then
+                if (Method is TLapeTree_InternalMethod) and (TLapeTree_InternalMethod(Method).ForceParam and (Tokenizer.Tok in ParserToken_Types + [tk_Identifier])) then
```

Comment:
Do not change this, and instead add semicolon to `ReturnOn` in `ParseStatement()`

```diff
--- a/lpcompiler.pas
+++ b/lpcompiler.pas
@@ -2868,7 +2868,7 @@ begin
     end
   else if (not (Tokenizer.Tok in ParserToken_BlockEnd)) then
   begin
-    Result := ParseExpression([], False);
+    Result := ParseExpression(ExprEnd, False);
     try
       ParseExpressionEnd(ExprEnd);
     except
```

Can you confirm whether this works?